kart/diff_structs.py

```diff
@@ -221,7 +223,7 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def ensure_child_type(self, key, value):
-        if type(value) != self.child_type:
+        if not isinstance(value, self.child_type):
             raise TypeError(
                 f"{type(self).__name__} accepts children of type {self.child_type.__name__} "
```

Comment:
this error message probably doesn't work correctly when `child_type` is a tuple of types